### ISSUE TITLE
(typescript) Optional type fields are not highlighted

### ISSUE THREAD

**Issue by ivanabovin** [highlightjs/highlight.js#4132]
**Describe the issue**
Optional type fields are not highlighted in `typescript`. They remains plain text.

**Which language seems to have the issue?**
`typescript`

**Are you using `highlight` or `highlightAuto`?**
`highlight`

```js
hljs.highlight(code, { language: 'typescript' }).value
```

**Sample Code to Reproduce**
https://jsfiddle.net/a28rv4ju/

```typescript
type User = {
  id: number;
  name?: string;
}
```

**Expected behavior**
Both `id` and `name` field names are highlighted.

**Actual behavior**
Only `id` field name is highlighted.

**Additional context**
`highlight.js@11.10.0`

**Comment by ngrbalu** [comment]
@ivanabovin  I can work on this. Please assign it to me.

**Comment by ivanabovin** [comment]
> @ivanabovin I can work on this. Please assign it to me.

I'm afraid I have no rights to do this

**Comment by joshgoebel** [comment]
Is this not fixed by #4114?  Can you please confirm this bug still exists in `main`.

**Comment by joshgoebel** [comment]
Not sure, releases are slower these days.  

I'm going to close this as fixed by #4114 for now.
